File: Dal/Models/Presentation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BPGezinswetenschappen.DAL.Models;

namespace DAL.Models
{
    public class Presentation
    {
        public int PresentationId { get; set; }
        public int StudentId { get; set; }
        public User Student { get; set; } = null!;

        public int CoachId { get; set; }
        public User Coach { get; set; } = null!;

        public int? ExpertId { get; set; }
        public User? Expert { get; set; } = null!;

        public ICollection<PresentationSlot> Slots { get; set; } = new List<PresentationSlot>();

    }
}

File: Client/src/app/shared/models/presentation.ts
import { Slot } from './slot';
import { User, UserReadDto } from './user';

// Note: To avoid circular reference issues, use 'any' for User and PresentationSlot for now.
export interface Presentation {
  presentationId: number;
  studentId: number;
  student: User;
  coachId: number;
  coach: User;
  expertId?: number;
  expert?: User;
  slots: Slot[];
}
